bdemeta/__main__.py

```diff
@@ -85,18 +100,9 @@ def make_resolver(config_path_str: str,
     except FileNotFoundError:
         raise NoConfigError(config_path)
 
-    result = []
-    for root in config['roots']:
-        path = pathlib.Path(root)
-        if path.is_absolute():
-            result.append(path)
-        else:
-            result.append(config_dir/path)
-    config['roots'] = result
-
-    for root in config['roots']:
-        if not root.is_dir():
-            raise InvalidPathError(root)
+    config['roots'] = normalize_roots(config['roots'], config_dir)
+    if 'conan_roots' in config:
+        config['conan_roots'] = normalize_roots(config['conan_roots'], config_dir)
```

Comment:
Based on my understanding of the change in `bdemeta/resolver.py`, it looks like the logic is: "if we identify this name as a valid target but its root is in `conan_roots`, it should be provided by a target named `CONAN_PKG::<name>`".

Would it be simpler implementation-wise (though more verbose configuration-wise) to have a `conan_targets` configuration key where we list all the targets that should come from Conan?

Reply:
> Would it be simpler implementation-wise (though more verbose configuration-wise)...

Just to be clear, I'm not sure if the above trade-off is a good one, but I wanted to hear your thoughts.

Reply:
It seems to me that this approach would be worth the maintenance burden if only some of the root's targets were provided by Conan.

Currently, there may be need to patch a given root so that it is compatible with bdemeta. I reckon this can be avoided if Conan provides all the root's targets, making the root "transparent" to bdemeta.